src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copy = Object.assign({}, state);
+  const result = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        for (const key in extraData) {
+          copy[key] = extraData[key];
+        };
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          if (copy.hasOwnProperty(key)) {
+            delete copy[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const key in copy) {
+          delete copy[key];
+        }
```

Comment:
It just copy, so we don't need to use loop
```suggestion
        copy = {};
```